### PR TITLE
[hail] Add TableAggregate simplify rules

### PR DESCRIPTION
**Pull request by tpoterba**
Better IR for:

```
ht = mt.cols()
ht.aggregate(...)
```

Instead of a `TableAgg(TableKeyBy(TableParallelize(x)))`, we get an `ArrayAgg(x)`.

### CODE REVIEW

File: hail/src/main/scala/is/hail/expr/ir/Exists.scala [hunk 1]
```diff
@@ -68,6 +68,25 @@ object ContainsAgg {
   })
 }
 
+object AggIsCommutative {
+  def apply(op: AggOp): Boolean = op match {
+    case Take() | Collect() | PrevNonnull() | TakeBy() => true
+    case _ => false
```
Comment (chrisvittal) [inline hail/src/main/scala/is/hail/expr/ir/Exists.scala]: this is backwards

File: hail/src/main/scala/is/hail/expr/ir/Exists.scala [hunk 2]
```diff
@@ -68,6 +68,25 @@ object ContainsAgg {
...
+  }
+}
+
+object ContainsNonCommutativeAgg {
```
Comment (chrisvittal) [inline hail/src/main/scala/is/hail/expr/ir/Exists.scala]: or maybe just confusing?

File: hail/src/main/scala/is/hail/expr/ir/Exists.scala (hunk 1 above)
Comment (tpoterba) [reply hail/src/main/scala/is/hail/expr/ir/Exists.scala]: oops yes

**Comment by tpoterba** [comment]
stacked on #7163

**Comment by tpoterba** [comment]
addressed your comment (good catch). I commented out a simplify rule that is currently invalid, but will be possible to introduce soon.

**Pull request merged**